packages/suite/src/reducers/wallet/sendFormReducer.ts

```diff
@@ -5,6 +5,12 @@ import { Action } from 'src/types/suite';
 import { FormState, PrecomposedTransactionFinal, TxFinalCardano } from 'src/types/wallet/sendForm';
 import { accountsActions } from '@suite-common/wallet-core';
 
+// Deep-cloning to prevent buggy interaction between react-hook-form and immer, see https://github.com/orgs/react-hook-form/discussions/3715#discussioncomment-2151458
+// Otherwise, whenever the outputs fieldArray is updated after the form draft or precomposedForm is saved, there is na error:
+// TypeError: Cannot assign to read only property of object '#<Object>'
+// This might not be necessary in the future when the dependencies are upgraded.
+const deepCopy = (obj: object) => JSON.parse(JSON.stringify(obj));
```

Comment:
This logic is used on many places within the project, maybe move it to some utils package?

Reply:
It [exists](https://github.com/trezor/trezor-suite/blob/0b8580a7bc41411d5a377a8b271f2204ba115bb2/packages/connect/src/utils/objectUtils.ts#L3) there already. And I once [tried to unify it](https://github.com/trezor/trezor-suite/pull/6071) and use a more modern solution.

Reply:
We had this function defined not once, but three times. I unified it and moved it to `@trezor/utils`. I also deleted some unused utils in `@trezor/connect` while I was at it. 9001fbab1046fe5fb20252b53885d190fb14fba7 🧹 

@dahaca, I read [this article](9001fbab1046fe5fb20252b53885d190fb14fba7) which recommends sticking with `JSON.parse` unless `structuredClone` features are required - because it performs better.